src/lib/passes/passmanager.cpp

```diff
@@ -0,0 +1,98 @@
+
+#include "coreir-pass/passmanager.h"
+
+using namespace CoreIR;
+
+void PassManager::addPass(Pass* p, uint ordering) {
+  if (passOrdering.count(ordering)==0) {
+    passOrdering[ordering] = unordered_map<uint,vector<Pass*>>();
+  }
+  passOrdering[ordering][p->getKind()].push_back(p);
+}
+
+
+//This does do pipelineing (Only one namespace!)
+bool PassManager::runNamespacePass(vector<Pass*>& passes) {
+  bool modified = false;
+  for (auto npass : passes) {
+    assert(isa<NamespacePass>(npass));
+    modified |= cast<NamespacePass>(npass)->runOnNamespace(this->ns);
+  }
+  instanceGraphStale = modified;
+  return modified;
+}
+
+//This does do pipelineing
+bool PassManager::runModulePass(vector<Pass*>& passes) {
+  bool modified = false;
+  for (auto modmap : ns->getModules()) {
+    Module* m = modmap.second;
+    for (auto mpass : passes) {
+      assert(isa<ModulePass>(mpass));
+      modified |= cast<ModulePass>(mpass)->runOnModule(m);
+    }
+  }
+  return modified;
+}
+
+
+//Does not do pipelineing
+bool PassManager::runInstanceGraphPass(vector<Pass*>& passes) {
+  
+  if (this->instanceGraphStale) {
+    instanceGraph.clear();
+    instanceGraph.construct(ns);
+  }
+  
+  bool modified = false;
+  for (auto igpass : passes) {
+    assert(isa<InstanceGraphPass>(igpass));
+    
+    for (auto node : this->instanceGraph.getSortedNodes()) {
+      modified |= cast<InstanceGraphPass>(igpass)->runOnInstanceGraphNode(*node);
+    }
+  }
+  return modified;
+}
+
+std::map<uint,unordered_map<uint,vector<Pass*>>> passOrdering;
```

Comment:
Just looked, it's defined in the PassManager declaration so this should probably be removed?